manifests/cf-manifest/scripts/generate-isolation-segment-ops-file.rb

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env ruby
+
+require "English"
+require "optparse"
+require "yaml"
+
+isolation_segment = YAML.safe_load(STDIN.read)
+
+options = {}
+OptionParser.new { |opts|
+  opts.on(
+    "-dPATH", "--isolation-segment-definition PATH",
+    "Path to definition for isolation segment"
+  ) do |d|
+    options[:definition_path] = d
+  end
+}.parse! ARGV
+
+abort "--isolation-segment-definition missing" if options[:definition_path].nil?
+
+seg_def = YAML.safe_load(File.read(options[:definition_path]))
+
+%w[
+  isolation_segment_name
+  number_of_cells
+]
+  .select { |var_name| seg_def[var_name].nil? }
+  .each { |var_name| abort "Could not find #{var_name}" }
+
+name = "diego-cell-iso-seg-#{seg_def['isolation_segment_name']}"
+isolation_segment["name"] = name
+
+isolation_segment["instances"] = seg_def["number_of_cells"]
+
+unless seg_def["isolation_segment_size"].nil?
+  vm_type = isolation_segment["vm_type"]
+
+  memory_capacity = isolation_segment
+    .fetch("jobs")
+    .find { |job| job["name"] == "rep" }
+    .dig("properties", "diego", "executor", "memory_capacity_mb")
+
+  case seg_def["isolation_segment_size"]
+  when "small"
+    memory_capacity /= 2.0
+    vm_type = "small_cell"
+  else
+    raise "Unknown isolation_segment_size #{seg_def['isolation_segment_size']}"
+  end
+
+  isolation_segment["vm_type"] = vm_type
+
+  isolation_segment
+    .fetch("jobs")
+    .find { |job| job["name"] == "rep" }[
+    "properties"][
+    "diego"][
+    "executor"][
+    "memory_capacity_mb"] = memory_capacity
+end
+
+if seg_def["restricted_egress"]
+  isolation_segment["jobs"] << {
+    "name" => "coredns",
+    "release" => "observability",
+    "properties" => { "corefile" =>
+      <<~COREFILE,
+      (common) {
+        health :8054
+        ready
+        log
+        prometheus :9153
+        bind 169.254.0.3
+      }
+
+      buildpacks.cloudfoundry.org {
+        import common
+        forward . 169.254.0.2:53
+      }
+
+      apps.internal {
+        import common
+        forward . 169.254.0.2:53
+      }
+      COREFILE
+    },
+  }
+
+  isolation_segment
+    .fetch("jobs")
+    .find { |job| job["name"] == "silk-cni" }[
+    "properties"][
+    "dns_servers"] = ["169.254.0.3"]
+
+  isolation_segment
+    .fetch("jobs")
+    .find { |job| job["name"] == "silk-cni" }[
+    "properties"][
+    "deny_networks"] = {
+      "running" => %w[
```

Comment:
We should note why and what these networks are covering. CIDR blocks ending in `/3` and `/2` aren't that common I don't think?